### PR TITLE
add test for adding feature in transaction mode with default value

### PR DESCRIPTION
**Pull request by 3nids**
add a test to detect failure on adding feature with default value in transaction mode.
should fail.

### CODE REVIEW

File: tests/src/python/test_qgsdefaultvaluetransaction.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        self.rollbackTransaction()
+
+    def test_transaction_add_feature_with_default_value(self):
+        """
+        Check if a feature can be added programmaticaly with a default value in transaction mode
+        """
+
+        with edit(self.vl):
+            f = QgsFeature()
+            init_fields = self.vl.dataProvider().fields()
```
Comment (m-kuhn) [inline tests/src/python/test_qgsdefaultvaluetransaction.py]: Just for good practice: shouldn't that be `self.vl.fields()`?

Comment (m-kuhn) [reply tests/src/python/test_qgsdefaultvaluetransaction.py]: Oh, and I think you can directly put that in the `f = QgsFeature( fields )` constructor

**Comment by 3nids** [comment]
@m-kuhn test is happily failing. I guess I'll wait for the fix to merge?

**Comment by m-kuhn** [comment]
You will need to use `QgsVectorLayerUtils.createFeature` to have default values. I assume that's on purpose (cc @nyalldawson)

**Pull request closed without merging**